src/composables/useFlawAffectsModel.ts

```diff
@@ -266,6 +270,34 @@ export function useFlawAffectsModel(flaw: Ref<ZodFlawType>) {
         });
       }
     }
+
+    if (Object.keys(affectCvssToDelete.value).length > 0) {
+      let cvssScoresRemovedCount = 0;
+      let affectWithRemovedCvssCount = 0;
+      try {
+        for (const affectId of Object.keys(affectCvssToDelete.value)) {
+          const affect = flaw.value.affects.find(affect => affect.uuid === affectId);
+          if (affect) {
+            const cvssId = affectCvssToDelete.value[affectId];
+            if (cvssId) {
+              await deleteAffectCvssScore(affect.uuid as string, cvssId);
+              cvssScoresRemovedCount++;
+            }
+          }
+        }
+        affectWithRemovedCvssCount++;
+      } catch (error) {
+        console.error('useFlawAffectsModel::saveAffects() Error removing CVSS score(s):', error);
+      }
```

Comment:
Using `await` inside a loop will "block" on each iteration, meaning that if the `deleteAffectCvssScore` takes 1 second to complete and we have 5 cvss, this will take 5 seconds as each request will be made sequentially.

We could use `Promise.allSettled` to improve this
```suggestion
      let affectWithRemovedCvssCount = 0;
      let cvssScoresRemovedCount = 0;
      const cvssScoresToDelete = [];
      try {
        for (const affectId of Object.keys(affectCvssToDelete.value)) {
          const affect = flaw.value.affects.find(affect => affect.uuid === affectId);
          if (affect) {
            const cvssId = affectCvssToDelete.value[affectId];
            if (cvssId) {
               cvssScoresToDelete.push(deleteAffectCvssScore(affect.uuid as string, cvssId));
            }
          }
        }
        cvssScoresRemovedCount = (await Promise.allSettled(cvssScoresToDelete)).filter(({ status }) => status === 'fulfilled').length;
        affectWithRemovedCvssCount++;
      } catch (error) {
        console.error('useFlawAffectsModel::saveAffects() Error removing CVSS score(s):', error);
      }
```

Reply:
Other than that I think this looks good

Reply:
Suggestion implemented. Thank you!